5-infrastructure/modules/base-env/main.tf

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/******************************************
+  Auto Find Environment Projects
+*****************************************/
+
+locals {
+  auto_gke_project_id        = length(data.google_projects.gke_project.projects) != 0 ? lookup(element(lookup(data.google_projects.gke_project, "projects", ""), 0), "project_id", "") : ""
```

Comment:
What are `auto_gke_project_id`? We should explicitly get project id's as user inputs

Reply:
if 4-projects is used to deploy, without changing the settings there it will look for the naming convention to auto find the projects in the parent folder. 
Can take it out though it is only used if the user does not give the project id themselves

Reply:
Yes lets keep it simple and require user input